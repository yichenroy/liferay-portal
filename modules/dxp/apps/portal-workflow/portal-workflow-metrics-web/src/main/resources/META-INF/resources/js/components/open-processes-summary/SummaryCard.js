/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import React from 'react';

/**
 * @class
 * @memberof open-processes-summary
 */
export default class SummaryCard extends React.Component {
	render() {
		const {description, total} = this.props;

		return (
			<div
				className="summary-card border col-2"
				style={{marginLeft: '16px'}}
			>
				<span className="regular-text semi-bold text-secondary">
					{description}
				</span>

				<div className="">
					<span
						className="font-weight-normal"
						style={{fontSize: '2.5rem'}}
					>
						{total}
					</span>

					<span className="regular-text text-secondary">
						{Liferay.Language.get('items')}
					</span>
				</div>
			</div>
		);
	}
}
