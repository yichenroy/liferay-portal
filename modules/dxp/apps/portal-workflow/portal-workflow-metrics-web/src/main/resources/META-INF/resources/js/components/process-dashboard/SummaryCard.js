import autobind from 'autobind-decorator';
import getCN from 'classnames';
import Icon from '../../shared/components/Icon';
import React from 'react';

const CLASSNAME = 'process-dashboard-summary-card';

export default class SummaryCard extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			inOver: false
		};
	}

	@autobind
	handleMouseOver() {
		this.setState({ inOver: !this.state.inOver });
	}

	@autobind
	handleMouseOut() {
		this.setState({ inOver: !this.state.inOver });
	}

	render() {
		const {
			elementClasses,
			iconColor,
			iconName,
			percentage,
			title,
			total
		} = this.props;

		const classes = getCN(CLASSNAME, elementClasses);

		const { inOver } = this.state;

		return (
			<a
				className={classes}
				href={'#'}
				onMouseOut={this.handleMouseOut}
				onMouseOver={this.handleMouseOver}
			>
				<div>
					<div className={'header'}>
						{iconName && (
							<Icon
								elementClasses={['iconBadge', `icon-${iconColor}`, 'mr-3']}
								iconName={iconName}
							/>
						)}
						{title}
					</div>
					<div className={'body'}>{total}</div>
					<div className={'footer'}>
						{percentage &&
							(!inOver ? (
								percentage
							) : (
								<span className={'highlight-hover'}>
									{Liferay.Language.get('see-items')}
								</span>
							))}
					</div>
				</div>
			</a>
		);
	}
}