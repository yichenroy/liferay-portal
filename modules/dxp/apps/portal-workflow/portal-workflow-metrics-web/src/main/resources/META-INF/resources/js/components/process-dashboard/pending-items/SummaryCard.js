import { AppContext } from '../../AppContext';
import autobind from 'autobind-decorator';
import { ChildLink } from '../../../shared/components/router/routerWrapper';
import { formatNumber } from '../../../shared/util/numeral';
import { getPercentage } from '../../../shared/util/util';
import Icon from '../../../shared/components/Icon';
import React from 'react';

class SummaryCard extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			hovered: false
		};
	}

	get formattedPercentage() {
		const { total, totalValue, value } = this.props;

		if (!total) {
			const percentage = getPercentage(value, totalValue);

			return formatNumber(percentage, '0[.]00%');
		}

		return null;
	}

	get formattedValue() {
		const { value } = this.props;
		const formatThreshold = 9999;

		if (value > formatThreshold) {
			return formatNumber(value, '0[,0][.]0a');
		}

		return formatNumber(value, '0[,]0');
	}

	@autobind
	handleMouseOver(evt, callback) {
		this.setState({ hovered: true }, callback);
	}

	@autobind
	handleMouseOut(evt, callback) {
		this.setState({ hovered: false }, callback);
	}

	render() {
		const { defaultDelta } = this.context;
		const { hovered } = this.state;
		const { iconColor, iconName, processId, title, total, value } = this.props;

		const dashboardItemsPath = `/instances/${processId}/${defaultDelta}/1`;

		const disabled = !total && value === undefined;

		const disabledClassName = disabled ? 'disabled' : '';
		const disableRender = Component => !disabled && Component;

		const hoveredClassName = hovered ? 'highlight-hover' : '';
		const hoverRender = (Component, HoveredComponent) => (
			<span className={`${hoveredClassName} xsmall`}>
				{(hovered && HoveredComponent) || Component}
			</span>
		);

		return (
			<ChildLink
				className={`${disabledClassName} process-dashboard-summary-card`}
				onMouseOut={this.handleMouseOut}
				onMouseOver={this.handleMouseOver}
				to={dashboardItemsPath}
			>
				<div>
					<div className={'header'}>
						{iconName && (
							<span
								className={`bg-${iconColor}-light mr-3 sticker sticker-circle`}
							>
								<span className="inline-item">
									<Icon
										elementClasses={`text-${iconColor}`}
										iconName={iconName}
									/>
								</span>
							</span>
						)}

						<span>{title}</span>
					</div>

					{disableRender(
						<div className="body" title={value}>
							{this.formattedValue}
						</div>
					)}

					{disableRender(
						<div className="footer">
							{hoverRender(
								this.formattedPercentage,
								Liferay.Language.get('see-items')
							)}
						</div>
					)}
				</div>
			</ChildLink>
		);
	}
}

SummaryCard.contextType = AppContext;
export default SummaryCard;