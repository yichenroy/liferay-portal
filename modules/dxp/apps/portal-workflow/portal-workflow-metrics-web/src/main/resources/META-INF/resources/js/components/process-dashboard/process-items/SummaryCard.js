import {filterKeys, processStatusKeys} from '../instance-list/filterConstants';
import {AppContext} from '../../AppContext';
import autobind from 'autobind-decorator';
import {ChildLink} from '../../../shared/components/router/routerWrapper';
import {formatNumber} from '../../../shared/util/numeral';
import {getPercentage} from '../../../shared/util/util';
import Icon from '../../../shared/components/Icon';
import React from 'react';

class SummaryCard extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			hovered: false
		};
	}

	get formattedPercentage() {
		const {total, totalValue, value} = this.props;

		if (!total) {
			const percentage = getPercentage(value, totalValue);

			return formatNumber(percentage, '0[.]00%');
		}

		return null;
	}

	get formattedValue() {
		return formatNumber(this.props.value, '0[,0][.]0a');
	}

	getFiltersQuery() {
		const {completed, slaStatusFilter, timeRange} = this.props;

		const filterParams = {
			[filterKeys.processStatus]: [
				completed
					? processStatusKeys.completed
					: processStatusKeys.pending
			],
			[filterKeys.slaStatus]: [slaStatusFilter]
		};

		if (timeRange) {
			filterParams[filterKeys.completionPeriod] = timeRange.key;
		}

		return filterParams;
	}

	@autobind
	handleMouseOver(evt, callback) {
		this.setState({hovered: true}, callback);
	}

	@autobind
	handleMouseOut(evt, callback) {
		this.setState({hovered: false}, callback);
	}

	render() {
		const {
			completed,
			getTitle,
			iconColor,
			iconName,
			processId,
			slaStatusFilter,
			total,
			value
		} = this.props;
		const {defaultDelta} = this.context;
		const {hovered} = this.state;

		const disabled = !total && value === undefined;

		const disabledClassName = disabled ? 'disabled' : '';
		const disableRender = Component => !disabled && Component;

		const hoveredClassName = hovered ? 'highlight-hover' : '';
		const hoverRender = (Component, HoveredComponent) => (
			<span className={`${hoveredClassName} xsmall`}>
				{(hovered && HoveredComponent) || Component}
			</span>
		);

		const instancesListPath = `/instances/${processId}/${defaultDelta}/1`;
		const title = getTitle(completed);

		return (
			<ChildLink
				className={`${disabledClassName} process-dashboard-summary-card`}
				onMouseOut={this.handleMouseOut}
				onMouseOver={this.handleMouseOver}
				query={{filters: this.getFiltersQuery(slaStatusFilter)}}
				to={instancesListPath}
			>
				<div>
					<div className={'header'}>
						{iconName && (
							<span
								className={`bg-${iconColor}-light mr-3 sticker sticker-circle`}
							>
								<span className='inline-item'>
									<Icon
										elementClasses={`text-${iconColor}`}
										iconName={iconName}
									/>
								</span>
							</span>
						)}

						<span>{title}</span>
					</div>

					{disableRender(
						<div className='body' title={value}>
							{this.formattedValue}
						</div>
					)}

					{disableRender(
						<div className='footer'>
							{hoverRender(
								this.formattedPercentage,
								Liferay.Language.get('see-items')
							)}
						</div>
					)}
				</div>
			</ChildLink>
		);
	}
}

SummaryCard.contextType = AppContext;
export default SummaryCard;