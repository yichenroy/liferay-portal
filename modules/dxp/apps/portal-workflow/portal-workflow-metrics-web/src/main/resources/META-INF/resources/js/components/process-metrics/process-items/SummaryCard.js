import {AppContext} from '../../AppContext';
import {ChildLink} from '../../../shared/components/router/routerWrapper';
import {filterConstants} from '../instance-list/store/InstanceListStore';
import {formatNumber} from '../../../shared/util/numeral';
import {formatQueryDate} from '../util/timeRangeUtil';
import {getPercentage} from '../../../shared/util/util';
import Icon from '../../../shared/components/Icon';
import {processStatusConstants} from '../filter/store/ProcessStatusStore';
import React from 'react';

class SummaryCard extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			hovered: false
		};
	}

	get formattedPercentage() {
		const {total, totalValue, value} = this.props;

		if (!total) {
			const percentage = getPercentage(value, totalValue);

			return formatNumber(percentage, '0[.]00%');
		}

		return null;
	}

	get formattedValue() {
		return formatNumber(this.props.value, '0[,0][.]0a');
	}

	getFiltersQuery() {
		const {completed, slaStatusFilter, timeRange} = this.props;

		const filterParams = {
			[filterConstants.processStatus]: [
				completed
					? processStatusConstants.completed
					: processStatusConstants.pending
			],
			[filterConstants.slaStatus]: [slaStatusFilter]
		};

		if (timeRange) {
			filterParams[filterConstants.timeRange] = [timeRange.key];
			filterParams[filterConstants.timeRangeDateEnd] = formatQueryDate(
				timeRange.dateEnd
			);
			filterParams[filterConstants.timeRangeDateStart] = formatQueryDate(
				timeRange.dateStart
			);
		}

		return filterParams;
	}

	handleMouseOver(evt, callback) {
		this.setState({hovered: true}, callback);
	}

	handleMouseOut(evt, callback) {
		this.setState({hovered: false}, callback);
	}

	render() {
		const {
			completed,
			getTitle,
			iconColor,
			iconName,
			processId,
			slaStatusFilter,
			total,
			value
		} = this.props;
		const {defaultDelta} = this.context;
		const {hovered} = this.state;

		const disabled = !total && value === undefined;

		const disabledClassName = disabled ? 'disabled' : '';
		const disableRender = Component => !disabled && Component;

		const hoveredClassName = hovered ? 'highlight-hover' : '';
		const hoverRender = (Component, HoveredComponent) => (
			<span className={`${hoveredClassName} xsmall`}>
				{(hovered && HoveredComponent) || Component}
			</span>
		);

		const instancesListPath = `/instances/${processId}/${defaultDelta}/1`;
		const title = getTitle(completed);

		return (
			<ChildLink
				className={`${disabledClassName} process-dashboard-summary-card`}
				onMouseOut={this.handleMouseOut.bind(this)}
				onMouseOver={this.handleMouseOver.bind(this)}
				query={{filters: this.getFiltersQuery(slaStatusFilter)}}
				to={instancesListPath}
			>
				<div>
					<div className={'header'}>
						{iconName && (
							<span
								className={`bg-${iconColor}-light mr-3 sticker sticker-circle`}
							>
								<span className="inline-item">
									<Icon
										elementClasses={`text-${iconColor}`}
										iconName={iconName}
									/>
								</span>
							</span>
						)}

						<span>{title}</span>
					</div>

					{disableRender(
						<div className="body" title={value}>
							{this.formattedValue}
						</div>
					)}

					{disableRender(
						<div className="footer">
							{hoverRender(
								this.formattedPercentage,
								Liferay.Language.get('see-items')
							)}
						</div>
					)}
				</div>
			</ChildLink>
		);
	}
}

SummaryCard.contextType = AppContext;
export default SummaryCard;
