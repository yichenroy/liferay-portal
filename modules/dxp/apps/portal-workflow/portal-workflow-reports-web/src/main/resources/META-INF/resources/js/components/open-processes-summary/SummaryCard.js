import React from 'react';

export default class SummaryCard extends React.Component {
	render() {
		const {description, total} = this.props;

		return (
			<div className="summary-card border col-2" style={{marginLeft: '16px'}}>
				<span className="regular-text semi-bold text-secondary">
					{description}
				</span>
				<div className="">
					<span className="font-weight-normal" style={{fontSize: '2.5rem'}}>
						{total}
					</span>
					<span className="regular-text text-secondary"> {'items'}</span>
				</div>
			</div>
		);
	}
}