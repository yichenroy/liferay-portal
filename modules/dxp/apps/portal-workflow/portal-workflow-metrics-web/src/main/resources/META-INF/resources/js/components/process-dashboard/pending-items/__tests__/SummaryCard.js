import React from 'react';
import { MockRouter as Router } from '../../../../test/mock/MockRouter';
import SummaryCard from '../SummaryCard';

test('Should format value for values with more than 4 digits', () => {
	const props = {
		iconColor: 'danger',
		iconName: 'exclamation-circle',
		processId: 12345,
		title: 'Overdue',
		total: false,
		totalValue: 858000,
		value: 156403
	};

	const component = mount(
		<Router>
			<SummaryCard {...props} />
		</Router>
	);

	const instance = component.find(SummaryCard).instance();

	expect(instance.formattedValue).toEqual('156.4K');
});

test('Should not format value for values with 4 or less digits', () => {
	const props = {
		iconColor: 'danger',
		iconName: 'exclamation-circle',
		processId: 12345,
		title: 'Overdue',
		total: false,
		totalValue: 3500,
		value: 3000
	};

	const component = mount(
		<Router>
			<SummaryCard {...props} />
		</Router>
	);

	const instance = component.find(SummaryCard).instance();

	expect(instance.formattedValue).toEqual('3,000');
});

test('Should render component', () => {
	const props = {
		iconColor: 'success',
		iconName: 'check-circle',
		processId: 12345,
		title: 'On Time',
		total: false,
		totalValue: 55,
		value: 31
	};

	const component = mount(
		<Router>
			<SummaryCard {...props} />
		</Router>
	);

	expect(component).toMatchSnapshot();
});

test('Should test mouse over', () => {
	const props = {
		iconColor: 'success',
		iconName: 'check-circle',
		processId: 12345,
		title: 'On Time',
		total: false,
		totalValue: 55,
		value: 31
	};

	const component = mount(
		<Router>
			<SummaryCard {...props} />
		</Router>
	);

	const instance = component.find(SummaryCard).instance();

	instance.handleMouseOver(null, () => {
		expect(instance.state.hovered).toEqual(true);
	});

	instance.handleMouseOut(null, () => {
		expect(instance.state.hovered).toEqual(false);
	});
});