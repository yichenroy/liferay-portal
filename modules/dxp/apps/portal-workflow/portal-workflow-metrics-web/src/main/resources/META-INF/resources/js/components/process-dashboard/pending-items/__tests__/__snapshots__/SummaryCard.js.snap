// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test mouse over 1`] = `
<MockRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/",
          "search": "?backPath=%2F",
        },
        Object {
          "pathname": "/slas/1234",
          "search": "?backPath=%2Fprocesses",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/",
              "search": "?backPath=%2F",
            },
            Object {
              "hash": "",
              "pathname": "/slas/1234",
              "search": "?backPath=%2Fprocesses",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 3,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SummaryCard
        elementClasses="test"
        iconColor="black"
        iconName="test"
        percentage={10}
        title="test"
        total={10}
      >
        <withRouter()
          className="process-dashboard-summary-card test"
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          to="/instances/undefined/20/1"
        >
          <Route>
            <Component
              className="process-dashboard-summary-card test"
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "canGo": [Function],
                  "createHref": [Function],
                  "entries": Array [
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    },
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "?backPath=%2F",
                    },
                    Object {
                      "hash": "",
                      "pathname": "/slas/1234",
                      "search": "?backPath=%2Fprocesses",
                    },
                  ],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "index": 0,
                  "length": 3,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "match": Object {
                      "params": Object {
                        "page": 1,
                        "search": undefined,
                        "sort": undefined,
                      },
                    },
                    "pathname": "/processes/1/10/title%3Aasc",
                    "search": "?backPath=%2F",
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "match": Object {
                    "params": Object {
                      "page": 1,
                      "search": undefined,
                      "sort": undefined,
                    },
                  },
                  "pathname": "/processes/1/10/title%3Aasc",
                  "search": "?backPath=%2F",
                }
              }
              match={
                Object {
                  "isExact": false,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              onMouseOut={[Function]}
              onMouseOver={[Function]}
              to="/instances/undefined/20/1"
            >
              <ChildLinkWrapper
                backPath="/"
                className="process-dashboard-summary-card test"
                currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "?backPath=%2F",
                      },
                      Object {
                        "hash": "",
                        "pathname": "/slas/1234",
                        "search": "?backPath=%2Fprocesses",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 3,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                onMouseOut={[Function]}
                onMouseOver={[Function]}
                to="/instances/undefined/20/1"
              >
                <Link
                  className="process-dashboard-summary-card test"
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  replace={false}
                  to={
                    Object {
                      "pathname": "/instances/undefined/20/1",
                      "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                    }
                  }
                >
                  <a
                    className="process-dashboard-summary-card test"
                    href="/instances/undefined/20/1?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                  >
                    <div>
                      <div
                        className="header"
                      >
                        <span
                          className="bg-black-light mr-3 sticker sticker-circle"
                        >
                          <span
                            className="inline-item"
                          >
                            <Icon
                              elementClasses="text-black"
                              iconName="test"
                            >
                              <svg
                                className="lexicon-icon lexicon-icon-test text-black"
                                focusable="false"
                                role="presentation"
                              >
                                <use
                                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#test"
                                />
                              </svg>
                            </Icon>
                          </span>
                        </span>
                        <span>
                          test
                        </span>
                      </div>
                      <div
                        className="body"
                      >
                        10
                      </div>
                      <div
                        className="footer"
                      >
                        10
                      </div>
                    </div>
                  </a>
                </Link>
              </ChildLinkWrapper>
            </Component>
          </Route>
        </withRouter()>
      </SummaryCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;
